tests/utils_tests/test_functional.py

```diff
@@ -47,43 +49,107 @@ def __bytes__(self):
         self.assertEqual(str(t), "Î am ā Ǩlâzz.")
         self.assertEqual(bytes(t), b"\xc3\x8e am \xc4\x81 binary \xc7\xa8l\xc3\xa2zz.")
 
+    def assertCachedPropertyWorks(self, attr, Class):
+        with self.subTest(attr=attr):
+            def get(source):
+                return getattr(source, attr)
+
+            # docstring should be preserved
+            self.assertEqual(get(Class).__doc__, 'Here is the docstring...')
+
+            inst = Class()
+
+            # check that it is cached
+            self.assertEqual(get(inst), get(inst))
+
+            # check that it returns the right thing
+            self.assertEqual(get(inst)[0], 1)
+
+            # check that state isn't shared between instances
+            inst2 = Class()
+            self.assertNotEqual(get(inst), get(inst2))
+
+            # check that it behaves like a property when there's no instance
+            self.assertIsInstance(get(Class), cached_property)
+
+            # check that 'other_value' didn't become a property
+            self.assertTrue(callable(inst.other_value))
+
     def test_cached_property(self):
         """
         cached_property caches its value and that it behaves like a property
         """
-        class A:
+        class Class:
```

Comment:
What's the rationale for changing this test class name? It looks like unnecessary noise to me.

Reply:
@adamchainz said to do it: https://github.com/django/django/pull/10033#discussion_r193897287